### ISSUE TITLE
v-model to a nested getter/setter computed property backed by vuex doesn't behave as expected

### ISSUE THREAD

**Issue by benmccallum** [vuejs/vue#8182]
### Version
2.5.2

### Reproduction link
[https://codepen.io/benmccallum/pen/NMzrMj](https://codepen.io/benmccallum/pen/NMzrMj)

### Steps to reproduce
* Type something into "firstName" field and see the value is output immediately to the right. This is as expected. (Uses suggested pattern for Vuex with v-model at https://vuex.vuejs.org/en/forms.html)

* See "lastName" field has an initial value of "[object Object]". This isn't expected.
* Type something into "lastName" field and see the value is NOT immediately output to the right. This isn't expected. 
* Tab out of the "lastName" field and notice now it will output to the right.

### What is expected?
1. On load, both inputs should be empty.
2. Typing into lastName should behave the same way as firstName (immediate output to side on 'input' event, not just focus out)

### What is actually happening?
1. lastName input has "[object Object]"
2. Typing into lastName does nothing to our output until you "focus out"

---
I'm basically wanting to nest custom getter/setter properties under a top-level computed property. I'm doing this because I'd like to nest all my forms fields under a computed property called "form" that returns an object where each property is itself an object defining the getter/setter to the Vuex store. 

I can potentially flatten these all with a map method that creates a computed property (getter/setter) for each field like `fieldName   "Field"` but the reason I'm trying to do this is to normalize a Vue plugin's usage across component state vs vuex state. 

Also, if this nesting means that the "form" property with nestings is recalculated every time a field changes, let me know as obviously that'd be a performance consideration that may drive me to the flat style and force me to figure out how I document the plugin's usage.

**Comment by yyx990803** [comment]
This is not supposed to work. Your `lastName.value` is an object with two functions named `get` and `set`, **they are not actually getters and setters**.

**Comment by benmccallum** [comment]
I did in fact try the native get and set syntax in my app, but I will try in my simpler repro and see if it works there. Thanks

**Comment by benmccallum** [comment]
Managed to get it working. I think in my real app I was having issues dealing with the scope being lost and not being able to access `this`, as you're only able to create the getters/setters with the proper function with method body syntax (no arrow functions). 

I've got it working in my repro and also used a `console.log/alert` to validate that the computed property isn't re-run every time the field is changed, which makes sense since it's just returning the getter/setter delegates.

Thanks for putting me back on track there Evan (@yyx990803)!
